### DISCUSSION TITLE
[SPOILER] How to solve challenge 7 properly

### DISCUSSION THREAD

**Discussion by iBeccoo** [Cyfrin/foundry-full-course-cu#311] [Q&A] [answered]
I had solve challenge 7 but i did with checking previous successful tx transaction by checking the state changes that was made by that transaction 
![image](https://github.com/Cyfrin/foundry-full-course-f23/assets/101125593/0fe4aec9-423d-4f69-a0a0-416a50209634)

But i dont think the way i solved the challenge aligned with the course. Is there any clue on how i suppose to solve the challenge ?

**Comment by ghost** [comment, marked as answer]
The proper way is solving it imo but probably you're right the challenges are a good chance to practice Solidity and Foundry. Personally I will not tell you the commands you should run in order to solve the challenge 😸, but I think I will give you enough hints.

[Cast](https://book.getfoundry.sh/cast/) provides everything you need to solve this challenge. You basically need to find [commands](https://book.getfoundry.sh/reference/cast/) to:
- read from the slot
- send a transaction to call `solveChallenge` with the correct inputs.

One last hint: the NFT contract has something that could cause a problem if you try to solve the challenge again.

**Reply by xueyuanhuang** [reply]
hey thanks, I though I have to read that three arguments, turn out just read from contract lol

**Comment by EngrPips** [comment]
Interesting.
